specification/machinelearningservices/data-plane/Microsoft.MachineLearningServices/preview/2019-05-01/execution.json

```diff
@@ -0,0 +1,1103 @@
+      }
+    },
+    "MpiConfiguration": {
+      "type": "object",
+      "properties": {
+        "processCountPerNode": {
+          "format": "int32",
+          "description": "Number of processes per node.",
+          "type": "integer",
+          "example": "2"
+        }
+      }
+    },
+    "HdiConfiguration": {
+      "type": "object",
+      "properties": {
+        "yarnDeployMode": {
+          "enum": [
+            "None",
+            "Client",
+            "Cluster"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "YarnDeployMode",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "ContainerInstanceConfiguration": {
+      "type": "object",
+      "properties": {
+        "region": {
+          "description": "Defaults to the region of the workspace.",
+          "type": "string",
+          "example": "eastus2"
+        },
+        "cpuCores": {
+          "format": "double",
+          "description": "Default size corresponds to the largest container supported in all regions.\r\nDetails: https://docs.microsoft.com/en-us/azure/container-instances/container-instances-quotas",
+          "type": "number",
+          "example": "2"
+        },
+        "memoryGb": {
+          "format": "double",
+          "description": "The memory available for the container instance.",
+          "type": "number",
+          "example": "3.5"
+        }
+      }
+    },
+    "PythonSection": {
+      "type": "object",
+      "properties": {
+        "interpreterPath": {
+          "description": "The python interpreter path. This is only used when user_managed_dependencies=True.",
+          "type": "string"
+        },
+        "userManagedDependencies": {
+          "description": "True means that AzureML reuses an existing python environment; False means that AzureML will create a python environment based on the Conda dependencies specification.",
+          "type": "boolean"
+        },
+        "condaDependencies": {
+          "type": "object"
+        },
+        "baseCondaEnvironment": {
+          "type": "string"
+        }
+      }
+    },
+    "DockerSection": {
+      "type": "object",
+      "properties": {
+        "baseImage": {
+          "description": "Base image used for Docker-based runs. Mutually exclusive with BaseDockerfile.",
+          "type": "string",
+          "example": "ubuntu:latest"
+        },
+        "baseDockerfile": {
+          "description": "Base Dockerfile used for Docker-based runs. Mutually exclusive with BaseImage.",
+          "type": "string",
+          "example": "FROM ubuntu:latest\r\nRUN echo \"Hello world!\""
+        },
+        "enabled": {
+          "description": "Set True to perform this run inside a Docker container.",
+          "type": "boolean",
+          "example": "True"
+        },
+        "sharedVolumes": {
+          "description": "Set False if necessary to work around shared volume bugs on Windows.",
```

Comment:
Can we double-check that this is the description we want in our public facing documentation?

Reply:
The bugs in question are Docker bugs, not Windows bugs -- that may be worth some rephrasing.

Reply:
corrected the comment

Reply:
done